### ISSUE TITLE
Stubbing multiple routes in one call

### ISSUE THREAD

**Issue by ckmoga** [microsoft/playwright#18684]
### Discussed in https://github.com/microsoft/playwright/discussions/18675

<div type='discussions-op-text'>

<sup>Originally posted by **ckmoga** November  9, 2022</sup>
I have page that makes 6 different API calls. I need to stub the responses of 3 of those API calls.
I could not find useful reference so I thought I would just intercept them one by one, like:

```
await context.route('**/api/books/', async route => {
    const response = await context.request.get('**/api/books/');
    expect(response.ok()).toBeTruthy();
    route.fulfill({
      response,
      body: JSON.stringify(mock1),
    });
  });

await context.route('**/api/authors/', async route => {
    const response = await context.request.get('**/api/authors/');
    expect(response.ok()).toBeTruthy();
    route.fulfill({
      response,
      body: JSON.stringify(mock2),
    });
  });
  
await context.route('**/api/category/', async route => {
    const response = await context.request.get('**/api/category/');
    expect(response.ok()).toBeTruthy();
    route.fulfill({
      response,
      body: JSON.stringify(mock3),
    });
  });  

```

The first intercept works but by the time it gets to the second one it says context is disposed:

apiRequestContext.get: Request context disposed.
The first intercept works but by the time it gets to the second one it says context is disposed:

apiRequestContext.get: Request context disposed.</div>

**Comment by yury-s** [comment]
This means that you test finishes before the second request is fulfilled. In that case playwright starts closing contexts while the interceptor is handling a request. You should make ensure that the test finishes only after all requests have been intercepted. It is up to you what condition to use for that but the easiest might be to [waitForResponse](https://playwright.dev/docs/api/class-page#page-wait-for-response) on the last url. Would that work for you?

**Comment by ckmoga** [comment]
Thanks @yury-s. I am already using ```await page.goto(url, { waitUntil: "networkidle" })``` to wait for the network to complete traffic.
Anyway, I have tried your suggestion and the result is the same. I think I am missing something because the flow isn't what I expect/wanted. It,
1. go to url
2. intercept first route
3. performs rest of test steps
4. tries to handle second route

My expectation/desire is step 4 comes before step 3. Whatever wait I put, it does not change the behavior. I event tried waitForResponse for all routes, with no effect.

**Comment by yury-s** [comment]
> My expectation/desire is step 4 comes before step 3. 

In that case one solution would be to have `await page.waitForResponse('**/api/authors/');` or something like that before getting to step 3.

Please share the code of the test if you want us to help, otherwise we can only guess what it does and what may be failing.

**Comment by pavelfeldman** [comment]
Closing as per above, please feel free to open a new issue if this does not cover your use case.
